include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -1067,19 +1473,24 @@ FEPointEvaluation<n_components, dim, spacedim, Number>::evaluate(
         mapping_info->get_mapping_data(current_cell_index, current_face_number);
 
       const std::size_t n_points = unit_points.size();
-      const std::size_t n_lanes  = VectorizedArray<Number>::size();
-      for (unsigned int i = 0, qb = 0; i < n_points; i += n_lanes, ++qb)
+      const std::size_t n_lanes =
+        internal::VectorizedArrayTrait<VectorizedArrayType>::width;
+      const std::size_t stride =
+        internal::FEPointEvaluation::VectorizedPointsTypeTraits<Number>::stride;
+
+      // loop over quadrature batches qb / points q
+      for (unsigned int qb = 0, q = 0; q < n_points; ++qb, q += n_lanes)
         {
           // compute
           const unsigned int n_shapes     = poly.size();
           const auto         val_and_grad = [&]() {
             if (polynomials_are_hat_functions)
               {
-                // convert to vectorized format
-                Point<dim, VectorizedArray<Number>> vectorized_points;
-                for (unsigned int j = 0; j < n_lanes && i + j < n_points; ++j)
+                // convert quadrature points to vectorized format
+                Point<dim, VectorizedArrayType> vectorized_points;
+                for (unsigned int v = 0; v < n_lanes && q + v < n_points; ++v)
                   for (unsigned int d = 0; d < dim; ++d)
-                    vectorized_points[d][j] = unit_points[i + j][d];
+                    vectorized_points[d][v] = unit_points[q + v][d];
```

Comment:
Reshuffling? We need to provide the vectorized quadrature points in the linear case in `evaluate()/integrate()` because the shape functions are evaluated on the fly in the specialized functions. The the general case with `p>1` we need to provide them during `reinit()` in `do_reinit()` to compute the `shapes`.